Automattic/pocket-casts-ios
# Fix for iPad crash on xcode 16 and iOS 18

| 📘 Part of: # |  
|:---:|

Fixes #2063

This PR avoid the crash resulting from the iPad Tab bar new design on iOS18 when using a TabViewController.

The way we are avoiding the crash is to force the TabBarController  to believe is always in the compact mode and keep using the regular tab bar on the bottom.

I tried to adapt to the new design but there are several issues with our current design of the app:

- Our mini-player is positioned above the tab bar ( this was reason for the crash because on the new design the tabbar is not in the same hierarchy as the mini-player)
- Our animations for the mini-player assume the tabbar is on the bottom
- The new tab bar design only allows to show 4 items on portrait mode and we currently 5 items so it show a chevron to display more options


## To test

- Using Xcode 16 beta 5 and an iOS 18 iPad device start the app
- Ensure that the app does not crash at startup
- Rotate the device and use the multiple split-view modes and check the app is working correctly
- Test on iOS 18 iPhone device to see if the tab bar shows correctly
- Repeat the tests on XCode 15.4 and iOS 17

## Checklist
